e2e/api/backend_test.go

```diff
@@ -237,8 +239,9 @@ func (s *APIBackendTestSuite) TestResetChainData() {
 	s.StartTestBackend(params.RinkebyNetworkID)
 	defer s.StopTestBackend()
 
-	// allow to sync for some time
-	s.EnsureNodeSync()
+	ctx, cancel := context.WithTimeout(context.Background(), DefaultNodeSyncTimeout)
+	defer cancel()
+	s.Nil(EnsureNodeSync(ctx, s.Backend.NodeManager()), "cannot ensure node synchronization")
```

Comment:
Why have you decided to control node sync timeout outside of the function responsible for ensuring the blockchain is synced?

Reply:
Has been after hint https://github.com/status-im/status-go/pull/410/files/cb67346740ef4784718331e2713a8f36caeb0df9#r145371642 to gain more flexibility. An optimisation for a so far not existing use case, yes.

Two solutions:

a) Keep it but also add a convenience wrapper.
b) Make the timeout handled internally again and add a duration instead of a context as argument.

In case of a possible node sync problematic outside of tests I would vote for (a), otherwise (b) is more straightforward and simpler. From my current insight into the project I would vote for (b).

Reply:
We can get rid of creating a context outside of `EnsureNodeSync` and still keep flexibility by passing `nil` as context and creating a default one in `EnsureNodeSync`. 

```go
s.NoError(EnsureNodeSync(nil, s.Backend.NodeManager()))

func EnsureNodeSync(ctx context.Context, nodeManager common.NodeManager) error {
  if ctx == nil {
    ctx, cancel = context.WithTimeout(context.Background(), DefaultNodeSyncTimeout)
    defer cancel()
  }
  // more code... 
}
```

Btw. it's better to use `s.NoError()` instead of `s.Nil()` as it's more descriptive if a function returns an `error`.

Reply:
Also a nice idea, yes, even if I'm no real fan of `nil` as explicit argument value. But if it's okay for all I'll change it.

Hint for `NoError()` is good, definitely more clear. Will also add such a statement to my (private) Tideland Assertion. 😉

Reply:
Let's judge from our use case. And our use case is: "just sync the blockchain and stop if it's taking too long", right? With "long" meaning, say, 20 minutes. And we only need it in tests. So let's just follow this requirement and implement a simple function which handles everything internally, I can't see now why we need context for that function.

Reply:
Yeah, you're probably right. This should be a general rule to avoid doing unnecessary features and not over-engineered. I am ok with removing context at all.

Sorry @themue for the confusion :(

Reply:
Feeling pretty good with almost any kind of simplification. So will change it now.